src/python/pants/backend/jvm/targets/scala_library.py

```diff
@@ -47,6 +51,8 @@ def __init__(self, java_sources=None, payload=None, **kwargs):
     })
     super().__init__(payload=payload, **kwargs)
 
+    self._scoverage_instance = ScoveragePlatform.global_instance()
```

Comment:
The target itself doesn't need to hold a reference to `_scoverage_instance`: can get it lazily from the singleton.